lib/versioning/Version.ts

```diff
@@ -83,6 +83,31 @@ export class Version {
         return `{ "isPHD": true, "versionId": "${versionId}" }`;
     }
 
+    /**
+     * Appends a key-value pair to a JSON object represented as a string. It adds
+     * a comma if the object is not empty (i.e., not just '{}'). It assumes the input
+     * string is formatted as a JSON object.
+     *
+     * @param {string} stringifiedObject The JSON object as a string to which the key-value pair will be appended.
+     * @param {string} key The key to append to the JSON object.
+     * @param {string} value The value associated with the key to append to the JSON object.
+     * @returns {string} The updated JSON object as a string with the new key-value pair appended.
+     * @example
+     * _jsonAppend('{"existingKey":"existingValue"}', 'newKey', 'newValue');
+     * // returns '{"existingKey":"existingValue","newKey":"newValue"}'
+     */
+    static _jsonAppend(stringifiedObject: string, key: string, value: string): string {
+        // stringifiedObject value has the format of '{...}'
+        let index = stringifiedObject.length - 2;
+        while (stringifiedObject.charAt(index--) === ' ');
+        const comma = stringifiedObject.charAt(index + 1) !== '{';
+        return (
+            `${stringifiedObject.slice(0, stringifiedObject.length - 1)}` + // eslint-disable-line
```

Comment:
Instead of a bare `// eslint-disable-line`, having the exact check code afterwards so we know exactly what we disable would be better (both to understand and to avoid disabling other potentially useful rules):

`// eslint-disable-line the-check-that-fails`

You can probably run the tests without that line to figure out what it is.